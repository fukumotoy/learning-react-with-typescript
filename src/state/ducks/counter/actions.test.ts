/* eslint-disable import/no-extraneous-dependencies */
import {
  increment,
  decrement,
  onClickIncrement,
  onClickDecrement,
  onClickIncrementIfOdd,
  onClickIncrementAsync,
} from './actions';

describe('Match Snapshot', (): void => {
  it('increment', (): void => {
    expect(increment()).toMatchSnapshot();
  });

  it('decrement', (): void => {
    expect(decrement()).toMatchSnapshot();
  });

  it('increment(5)', (): void => {
    expect(increment(5)).toMatchSnapshot();
  });

  it('decrement(5)', (): void => {
    expect(decrement(5)).toMatchSnapshot();
  });

  it('onClickIncrement', (): void => {
    expect(onClickIncrement()).toMatchSnapshot();
  });

  it('onClickDecrement', (): void => {
    expect(onClickDecrement()).toMatchSnapshot();
  });

  it('onClickIncrementIfOdd', (): void => {
    expect(onClickIncrementIfOdd()).toMatchSnapshot();
  });

  it('onClickIncrementAsync', (): void => {
    expect(onClickIncrementAsync()).toMatchSnapshot();
  });
});
