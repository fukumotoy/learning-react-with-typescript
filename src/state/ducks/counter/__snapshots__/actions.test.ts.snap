// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match Snapshot decrement 1`] = `
Object {
  "payload": 1,
  "type": "COUNTER/DECREMENT",
}
`;

exports[`Match Snapshot decrement(5) 1`] = `
Object {
  "payload": 5,
  "type": "COUNTER/DECREMENT",
}
`;

exports[`Match Snapshot increment 1`] = `
Object {
  "payload": 1,
  "type": "COUNTER/INCREMENT",
}
`;

exports[`Match Snapshot increment(5) 1`] = `
Object {
  "payload": 5,
  "type": "COUNTER/INCREMENT",
}
`;

exports[`Match Snapshot onClickDecrement 1`] = `
Object {
  "type": "USER_ACTION/ON_CLICK/DECREMENT",
}
`;

exports[`Match Snapshot onClickIncrement 1`] = `
Object {
  "type": "USER_ACTION/ON_CLICK/INCREMENT",
}
`;

exports[`Match Snapshot onClickIncrementAsync 1`] = `
Object {
  "type": "USER_ACTION/ON_CLICK/INCREMENT_ASYNC",
}
`;

exports[`Match Snapshot onClickIncrementIfOdd 1`] = `
Object {
  "type": "USER_ACTION/ON_CLICK/INCREMENT_IF_ODD",
}
`;
